### PR TITLE
use ApproxManiProds.skew instead

### PR DESCRIPTION
**Pull request by dehann**
waiting on:
- https://github.com/JuliaRegistries/General/pull/112652

### CODE REVIEW

File: src/factors/Inertial/IMUDeltaFactor.jl [hunk 1]
```diff
@@ -340,6 +326,9 @@ end
 IIF.getManifold(::IMUDeltaFactor) = IMUDeltaGroup()
 
 function IIF.preambleCache(fg::AbstractDFG, vars::AbstractVector{<:DFGVariable}, ::IMUDeltaFactor)
+    # FIXME, nsec should only contain fractional second information, i.e. < 1.0s.  Use together with `trunc(timestamp) + 1e-9*nsec`
```
Comment (Affie) [inline src/factors/Inertial/IMUDeltaFactor.jl:329]: That is not how I use it and not how its documented.

File: src/factors/Inertial/IMUDeltaFactor.jl [hunk 2]
```diff
@@ -5,20 +5,6 @@ using LinearAlgebra
 using DistributedFactorGraphs
 using Dates
 
-# FIXME use ApproxManifoldsProducts version instead
```
Comment (Affie) [inline src/factors/Inertial/IMUDeltaFactor.jl:8]: Why not move it to TU?

File: src/factors/Inertial/IMUDeltaFactor.jl [hunk 3]
```diff
@@ -324,7 +310,7 @@ Base.@kwdef struct IMUDeltaFactor{T <: SamplableBelief} <: AbstractManifoldMinim
     J_b::SMatrix{10,6,Float64} = zeros(SMatrix{10,6,Float64})
     # accelerometer bias, gyroscope bias 
     b::SVector{6, Float64} = zeros(SVector{6, Float64})
-    #optional raw measurements
+    #optional raw measurements, FIXME replace with BlobEntry -- dont do abstract types here, or raw data if not needed in hotloop
```
Comment (Affie) [inline src/factors/Inertial/IMUDeltaFactor.jl:313]: What abstract types?

Comment (dehann) [reply src/factors/Inertial/IMUDeltaFactor.jl:313]: does this not create an ablstract type raw_measurement::Union?

File: src/factors/Inertial/IMUDeltaFactor.jl (hunk 1 above)
Comment (dehann) [reply src/factors/Inertial/IMUDeltaFactor.jl:329]: Again, we should do what ROS does and not our own special thing.  By the way, the suggested code here works for both cases.  We should also update our documentation to reflect the same as ROS.

Comment (Affie) [reply src/factors/Inertial/IMUDeltaFactor.jl:329]: We had this discussion before and also didn't agree then.
If you want to do what ros does:
Split the timestamp in two 32 bit timestamps, one for your ns and the other for the second part. The 2 combined as a single 64bit timestamp is basically what you have here.

> By the way, the suggested code here works for both cases. We should also update our documentation to reflect the same as ROS.

I feel like a parrot, but it doesn't, I've had multiple cases where there are 2 different timestamps. Not all timestamps are synced, and not all timestamps are unix epoch.

Comment (dehann) [reply src/factors/Inertial/IMUDeltaFactor.jl:329]: What we should not do is say `getTimestamp(var)` is not actually the timestamp.

Perhaps we should rename `.nstime` field to: `variable.missionnanosec` as part of DFG v1.  The suggested comments in code here already support both ("mission time" and ROS standard) cases where the full mission time is in the `nanosecond` field, or is split with the `timestamp` field.

This way the poweruser would know that `timestamp` and `missionnanosec` are equivalent.  They can then have different start times also, namely actual calendar `datetime+timezone`, and `0ns`.  They can then be made equivalent with 
`datetime2unix(timestamp[1]) + missionnanosec*1e-9`
but this sum is the really large number where Float64 risk of precision occurs.  That would be in line with `DateTime` being millisecond only.

We should obviously be using Float64 for this, so don't mind ROS' use of Float32.  Remember also that much of the web still uses 32 for many of its workings.

The suggested `missionnanosec` approach seems to solve all the problems?

Comment (dehann) [reply src/factors/Inertial/IMUDeltaFactor.jl:329]: https://github.com/JuliaRobotics/DistributedFactorGraphs.jl/issues/1087

Comment (dehann) [reply src/factors/Inertial/IMUDeltaFactor.jl:329]: > 32 bit timestamps

I mean that we stay with 64 bit (obviously).

Note some web stacks only support 32 bit at this time.  Some of the json libraries was the example we bumped into before.

Comment (dehann) [inline src/factors/Inertial/IMUDeltaFactor.jl:329]:
```suggestion
    # FIXME, change to `.missionnanosec` which can be used together with `trunc(timestamp) + 1e-9*nsec`.  See DFG #1087
```

Comment (dehann) [reply src/factors/Inertial/IMUDeltaFactor.jl]: updated the `FIXME` comment to be inline with `missionnanosec`.

File: src/factors/Inertial/IMUDeltaFactor.jl (hunk 2 above)
Comment (Affie) [reply src/factors/Inertial/IMUDeltaFactor.jl:8]: probably because TU will be deprecated?

File: src/factors/Inertial/IMUDeltaFactor.jl (hunk 3 above)
Comment (Affie) [reply src/factors/Inertial/IMUDeltaFactor.jl:313]: Both Nothing and IMUMeasurement are concrete types. Union{Nothing, T} are normally not a problem

**Pull request merged**